### ISSUE TITLE
Unable to use http_node / http_static endpoints with credentials

### ISSUE THREAD

**Issue by regystro** [hassio-addons/addon-node-red#200]
Hi.

I'm trying to set up an http in node endpoint with custom login credentials, but I can't get it to work. Either the configured credentials are not working or I'm forced to use Hassio admin credentials.

**Steps to reproduce:**

- Simple Node-RED http in node -> Method: GET, URL: /simpletest
- Add-on config:
  * Web interface port 1880
  * http_node + http_static (same user/password set for both)
    - http://\<mycustomduckdnshost>.org:1880/simpletest Result: login prompt. Neither http_node nor Hassio admin credentials work.
    - http://\<mycustomduckdnshost>.org:1880 Result: login prompt. Hassio admin login works. Shows Node-RED admin UI

Even tried changing `settings.js` httpAdminRoot: '/admin'. Same happens, just using http://\<mycustomduckdnshost>.org:1880/admin instead.

I'm testing always clearing Local Storage auth credentials (https://github.com/hassio-addons/addon-node-red/issues/190#issuecomment-483463573)

Also tried adding to configuration.yaml as stated here (https://github.com/hassio-addons/addon-node-red/issues/184#issuecomment-483230107)
```
  auth_providers:
    - type: homeassistant

```
I was using reverse proxy and SSL at first, but now I am testing with plain HTTP and port. Also trying LAN access using HASSIO IP, same result.

I've only been able to access the endpoint this way, but don't want to use Hassio credentials:
* http_node + http_static default config (no user/password):
  - http://\<mycustomduckdnshost>.org:1880/simpletest Result: login prompt. Hassio admin login works. Shows endpoint response
  - http://\<mycustomduckdnshost>.org:1880 Result: login prompt. Hassio admin login works. Shows Node-RED admin UI

Any clues?

**Comment by regystro** [comment]
By the way:
```
 Add-on version: 3.0.1
 You are running the latest version of this add-on.
 System: Raspbian GNU/Linux 9 (stretch)  (armv7 / raspberrypi3)
 Home Assistant version: 0.91.4
 Supervisor version: 160
```

**Comment by wootapa** [comment]
I'm experiencing the same issue. Custom username/password for http_node/http_static doesn't have any effect. Leaving login/password empty should leave endpoints public, but I still get prompted. Only hassio login seems to work (but not legacy_api_password).
My auth providers:
`auth_providers:
    - type: homeassistant
    - type: legacy_api_password
      api_password: !secret http_password`

**Comment by regystro** [comment]
> I'm experiencing the same issue. Custom username/password for http_node/http_static doesn't have any effect. Leaving login/password empty should leave endpoints public, but I still get prompted. Only hassio login seems to work (but not legacy_api_password).
> My auth providers:
> 
> ```yaml
> auth_providers:
>   - type: homeassistant
>   - type: legacy_api_password
>     api_password: !secret http_password
> ```

**Comment by sanderdw** [comment]
I have the same problem. I would like the admin section secured by HA auth and the http endpoints with separate or no authorization.

By setting the option "leave_front_door_open" it also disables the admin authentication, which i do not want.

**Comment by ciB89** [comment]
+1, used this to speak through IFTTT to my nodered. This only happened after disabling legacy api_password

**Comment by ciB89** [comment]
Have you tried adding the user/pw directly to the URL?

`https://_user_:_password_@xxxx.duckdns.org:1880/doSomething}` works for me. Although I'm not too sure if my http_node user and password is actually different to hassio one.

**Comment by sinclairpaul** [comment]
Can these endpoints all be hosted in the same folder i.e. https://myserver:noderedport/openwithoutauth/something?  If so @frenck is it worth creating a sub folder that isn't lua protected in Nginx?, happy to do the pull if this is the case, shouldn't impact anyone not using it as it would just 404.

**Comment by jinhong-** [comment]
> Have you tried adding the user/pw directly to the URL?
> 
> `https://user:password@xxxx.duckdns.org:1880/doSomething` works for me. Although I'm not too sure if my http_node user and password is actually different to hassio one.

Turns out that you need both. You need to have a set of home assistant credentials and you need to set the http_node credentials and ensure they are using the exact same username and password

**Comment by frenck** [comment]
This issue has been addressed in our development version and will be generally available in the next release.

**Comment by frenck** [comment]
Fixed in release v3.1.0.
Please be sure to read the release notes & documentation on how to use it on this add-on.

**Comment by regystro** [comment]
Thank you. I can confirm it works flawlessly.
